Analyzer/src/AnalyzeLepTrigger.cc

```diff
@@ -109,356 +97,158 @@ void AnalyzeLepTrigger::Loop(NTupleReader& tr, double weight, int maxevents, boo
 
-        if( runYear == "2017" ) {
+        } else if( runYear == "2017" ) {
             std::vector<std::string> myMuonTriggers         { "HLT_IsoMu24", "HLT_IsoMu27", "HLT_Mu50", "HLT_TkMu50" };
-            std::vector<std::string> myElectronTriggers     { "HLT_Ele35_WPTight_Gsf", "HLT_Photon200", "HLT_Ele115CaloIdVT_GsfTrkIdT" };
+            std::vector<std::string> myElectronTriggers     { "HLT_Ele35_WPTight_Gsf", "HLT_Photon200", "HLT_Ele115_CaloIdVT_GsfTrkIdT" };
+        } else if (runYear == "2018") {
+            std::vector<std::string> myMuonTriggers         { "HLT_IsoMu24", "HLT_IsoMu27", "HLT_Mu50" };
+            std::vector<std::string> myElectronTriggers     { "HLT_Ele35_WPTight_Gsf", "HLT_Photon200", "HLT_Ele115_CaloIdVT_GsfTrkIdT" };
         }
 
         bool passMuonTriggers       = passTriggerGeneral( myMuonTriggers, TriggerNames, TriggerPass );
         bool passElectronTriggers   = passTriggerGeneral( myElectronTriggers, TriggerNames, TriggerPass );
         
-        //Define offline baseline for one lepton analysis
-        bool passOfflineBaseline    = ( JetID && passMadHT && NGoodBJets_pt30 >= 1 && Mbl > 50 && Mbl < 250 && HT_trigger_pt30 > 300 );
-
         bool pass6JetCut            = ( NGoodJets_pt30 >= 6 );
         bool pass5JetCut            = ( NGoodJets_pt30 >= 5 );
         bool pass4JetCut            = ( NGoodJets_pt30 >= 4 );
         bool pass3JetCut            = ( NGoodJets_pt30 >= 3 );
         bool pass2JetCut            = ( NGoodJets_pt30 >= 2 );
         bool pass1JetCut            = ( NGoodJets_pt30 >= 1 ); //A little redundant because we require one good b jet but for symmetry reasons
 
-        bool atLeastOneGoodEl       = ( std::any_of( GoodElectrons.begin(), GoodElectrons.end(), [] ( bool boolEl ) { return boolEl; } ) );
-        bool atLeastOneGoodMu       = ( std::any_of( GoodMuons.begin(), GoodMuons.end(), [] ( bool boolMu ) { return boolMu; } ) );
-
-
-        //------------------------------------------------------
-        //-- Do the Electron Trigger Efficiency on the Single Muon Dataset
-        //------------------------------------------------------
+        //-----------------------------------------------------------------------
+        //-- Do the Electron Trigger Efficiency on the Single Muon Dataset and MC
+        //-----------------------------------------------------------------------
         
-        if( filetag == "2016_Data_SingleMuon" || filetag == "2017_Data_SingleMuon" ) {
-            
-            if( !passOfflineBaseline || !atLeastOneGoodMu ) continue;
+        if( (filetag.find("SingleMuon") != std::string::npos || runtype == "MC") ) {
 
-            bool foundMuonPt35      = false;
-            bool foundMuonPt40      = false;
+            bool atLeastOneGoodMu = ( std::any_of( GoodMuons.begin(), GoodMuons.end(), [] ( bool boolMu ) { return boolMu; } ) );
```

Comment:
You don't need to loop though the muons to see if there is at least one. You can just just the NGoodMuon variable.

Reply:
That wasn't me.

Reply:
But I fixed.